kratos/utilities/read_materials_utility.cpp

```diff
@@ -0,0 +1,265 @@
+            } else if (value.IsDouble()) {
+                p_prop->SetValue(variable, static_cast<int>(value.GetDouble()));
+            } else {
+                KRATOS_ERROR << "Check you write the value in a correct format: " << value << std::endl;
+            }
+        } else if(KratosComponents<Variable<array_1d<double, 3> > >::Has(variable_name)) {
+            const Variable<array_1d<double, 3>>& variable = KratosComponents<Variable<array_1d<double, 3>>>().Get(variable_name);
+            if (value.IsVector()) {
+                array_1d<double, 3> temp(3, 0.0);
+                const Vector& value_variable = value.GetVector();
+                const std::size_t iter_number = (3 < value_variable.size()) ? 3 : value_variable.size();
+                for (std::size_t index = 0; index < iter_number; index++)
+                    temp[index] = value_variable[index];
+                p_prop->SetValue(variable, temp);
+            } else {
+                KRATOS_ERROR << "Check you write the value in a correct format: " << value << std::endl;
+            }
+        } else if(KratosComponents<Variable<array_1d<double, 6> > >::Has(variable_name)) {
+            const Variable<array_1d<double, 6>>& variable = KratosComponents<Variable<array_1d<double, 6>>>().Get(variable_name);
+            if (value.IsVector()) {
+                array_1d<double, 6> temp(6, 0.0);
+                const Vector& value_variable = value.GetVector();
+                const std::size_t iter_number = (6 < value_variable.size()) ? 6 : value_variable.size();
+                for (std::size_t index = 0; index < iter_number; index++)
+                    temp[index] = value_variable[index];
+                p_prop->SetValue(variable, temp);
+            } else {
+                KRATOS_ERROR << "Check you write the value in a correct format: " << value << std::endl;
+            }
+        } else if(KratosComponents<Variable<Vector > >::Has(variable_name)) {
+            const Variable<Vector>& variable = KratosComponents<Variable<Vector>>().Get(variable_name);
+            if (value.IsVector()) {
+                p_prop->SetValue(variable, value.GetVector());
+            } else if (value.IsMatrix()) {
+                Vector temp;
+                const Matrix& value_variable = value.GetMatrix();
+                for (std::size_t index = 0; index < value_variable.size1(); index++)
+                    temp[index] = value_variable(index, 0);
+                p_prop->SetValue(variable, temp);
+            } else {
+                KRATOS_ERROR << "Check you write the value in a correct format: " << value << std::endl;
+            }
+        } else if(KratosComponents<Variable<Matrix> >::Has(variable_name)) {
+            const Variable<Matrix>& variable = KratosComponents<Variable<Matrix>>().Get(variable_name);
+            if (value.IsMatrix()) {
+                p_prop->SetValue(variable, value.GetMatrix());
+            } else if (value.IsVector()) {
```

Comment:
same as for Matrix

Reply:
I think it does not the coversion

https://github.com/KratosMultiphysics/Kratos/blob/c8cde04ff759696d3711ff69e7349f87e10ba4b4/kratos/includes/kratos_parameters.h#L366

Reply:
see comment above